activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1864,7 +1864,11 @@ def transform_select_hash_values(fields)
             end
           when String, Symbol
             arel_column(key.to_s) do
-              predicate_builder.resolve_arel_attribute(klass.table_name, key.to_s)
+              if (/(\w+)\s*\(/).match?(key.to_s)
+                Arel.sql(key.to_s)
```

Comment:
I think the point of the discussion in #45612 was that this does not happen automatically, but the user has to knowingly opt-in by calling `Arel.sql` themselves. 

So the user has to do 
```ruby
Post.joins(:comments).select(Arel.sql('COUNT(comments.id)') => :comment_count).take
```
to make it abundantly clear that they pass in raw sql.

Reply:
@fschwahn, then I see no point in this PR as user can simply write it in this way:
```ruby
Post.joins(:comments).select(Arel.sql("COUNT(comments.id)").as("comment_count")).take
```
with a current solution 🤔

Reply:
Yeah, I confirm that I don't feel confident allowing unmarked arbitrary SQL in there.

Also the implementation using a regexp to try to guess whether it's a function is not robust enough at all. Not all expressions are function calls, e.g. `Model.select('something + 1')` is a totally valid expression.

@simi if you still want the `Arel.sql` key thing, feel free to submit a PR and tag me.